app/services/discourse_subscriptions/campaign.rb

```diff
@@ -48,6 +50,27 @@ def create_campaign
 
     protected
 
+    def check_goal_status
+      original_status = SiteSetting.discourse_subscriptions_campaign_goal_met.dup
```

Comment:
I don't think you need to `.dup` this, it doesn't look like you are manipulating the setting value.

Reply:
We are later on L65 (depending on the value of `current_volume >= goal`). I assumed `.dup` was necessary since we're pointing to the value, but my Ruby knowledge isn't hyper deep.